engine/classes/Elgg/I18n/Translator.php

```diff
@@ -61,16 +61,24 @@ public function translate($message_key, $args = [], $language = "") {
 
 		$this->ensureTranslationsLoaded($language);
 
-		if (isset($GLOBALS['_ELGG']->translations[$language][$message_key])) {
-			$string = $GLOBALS['_ELGG']->translations[$language][$message_key];
+		$notice = '';
+		$string = $message_key;
 
-		} else if (isset($GLOBALS['_ELGG']->translations["en"][$message_key])) {
-			$string = $GLOBALS['_ELGG']->translations["en"][$message_key];
-			_elgg_services()->logger->notice(sprintf('Missing %s translation for "%s" language key', $language, $message_key));
+		foreach ([$language, 'en'] as $try_lang) {
+			if (isset($GLOBALS['_ELGG']->translations[$try_lang][$message_key])) {
+				$string = $GLOBALS['_ELGG']->translations[$try_lang][$message_key];
+				break;
+			} else {
+				$notice = sprintf(
+					'Missing %s translation for "%s" language key',
+					($try_lang === 'en') ? 'English' : $try_lang,
```

Comment:
Can we just always use the code? I don't think anyone would consider this part of the API.